src/main/java/io/horizen/lambo/CarRegistryApplicationState.java

```diff
@@ -1,39 +1,85 @@
 package io.horizen.lambo;
 
+import com.google.inject.Inject;
 import com.horizen.block.SidechainBlock;
 import com.horizen.box.Box;
 import com.horizen.proposition.Proposition;
 import com.horizen.state.ApplicationState;
 import com.horizen.state.SidechainStateReader;
 import com.horizen.transaction.BoxTransaction;
+import io.horizen.lambo.car.box.CarBox;
+import io.horizen.lambo.car.box.CarSellOrderBox;
+import io.horizen.lambo.car.services.CarInfoDBService;
+import io.horizen.lambo.car.transaction.CarDeclarationTransaction;
 import scala.util.Success;
 import scala.util.Try;
-
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+
 
-// There is no custom logic for Car registry State now.
-// TODO: prevent the declaration of CarBoxes which car information already exists in the previously added CarBoxes or CarSellOrderBoxes.
 public class CarRegistryApplicationState implements ApplicationState {
+	
+	private CarInfoDBService carInfoDbService;
+
+	@Inject
+	public CarRegistryApplicationState(CarInfoDBService carInfoDbService) {
+	    this.carInfoDbService = carInfoDbService;
+	}
+	
     @Override
     public boolean validate(SidechainStateReader stateReader, SidechainBlock block) {
-        return true;
+	    return true;
```

Comment:
We should also check that there are no multiple transactions declaring the same `VIN` inside the block.
In the `validate` method below we only check the uniqueness against state.